gcexport.py

```diff
@@ -652,28 +719,28 @@ def export_data_file(activity_id, activity_details, args, file_time, append_desc
         makedirs(directory)
 
     # timestamp as prefix for filename
-    if args.fileprefix > 0:
-        prefix = "{}-".format(start_time_locale.replace("-", "").replace(":", b"").replace(" ", "-"))
+    if args.fileprefix and args.fileprefix > 0:
+        prefix = "{}-".format(start_time_locale.replace("-", "").replace(":", "").replace(" ", "-"))
     else:
         prefix = ""
 
     fit_filename = None
     if args.format == 'gpx':
-        data_filename = directory + '/' + prefix + 'activity_' + activity_id + append_desc + '.gpx'
+        data_filename = directory + sep + prefix + 'activity_' + activity_id + append_desc + '.gpx'
         download_url = URL_GC_GPX_ACTIVITY + activity_id + '?full=true'
         file_mode = 'w'
     elif args.format == 'tcx':
-        data_filename = directory + '/' + prefix + 'activity_' + activity_id + append_desc + '.tcx'
+        data_filename = directory + sep + prefix + 'activity_' + activity_id + append_desc + '.tcx'
         download_url = URL_GC_TCX_ACTIVITY + activity_id + '?full=true'
         file_mode = 'w'
     elif args.format == 'original':
-        data_filename = directory + '/' + prefix + 'activity_' + activity_id + append_desc + '.zip'
+        data_filename = directory + sep + prefix + 'activity_' + activity_id + append_desc + '.zip'
         # TODO not all 'original' files are in FIT format, some are GPX or TCX...
-        fit_filename = directory + '/' + prefix + 'activity_' + activity_id + append_desc + '.fit'
+        fit_filename = directory + sep + prefix + 'activity_' + activity_id + append_desc + '.fit'
         download_url = URL_GC_ORIGINAL_ACTIVITY + activity_id
         file_mode = 'wb'
     elif args.format == 'json':
-        data_filename = directory + '/' + prefix + 'activity_' + activity_id + append_desc + '.json'
+        data_filename = directory + sep + prefix + 'activity_' + activity_id + append_desc + '.json'
```

Comment:
Here is a suggestion for the first case but the other should updated in the same way
```
        data_filename = join(directory, f'{prefix}activity_{activity_id}{append_desc}.gpx')
```
Also, `data_filename` should be replace with `data_filepath` due to using a directory path at the beginning.
And missing namespacing (`os.path.join`)

Reply:
In the PR #32 the need for more flexible filename patterns arose; I'd like to rewrite this code anyway

Reply:
Still it should be like this:
`data_filename = join(directory, prefix + 'activity_' + activity_id + append_desc + '.gpx')`

Other than that, I agree it should be rewritten one day ;)